aat/auth_test.go

```diff
@@ -192,3 +248,45 @@ func (ks *serverKeyStore) AuthRole(authid string) (string, error) {
 	}
 	return "user", nil
 }
+
+func (ks *serverKeyStore) AlreadyAuth(authid string, details wamp.Dict) bool {
+	// Verify that the request has been captured.
+	v, err := wamp.DictValue(details, []string{"transport", "auth", "request"})
+	if err != nil {
+		// Not authorized - pretend request is needed for some reason.
+		return false
+	}
+
+	v, err = wamp.DictValue(details, []string{"transport", "auth", "cookie"})
+	if err != nil {
+		// No tracking cookie - client not recognized, so not auth.
+		return false
+	}
+	cookie := v.(*http.Cookie)
+
+	// Tracking cookie matches cookie of previously good client.
+	if cookie.Value != ks.cookie.Value {
```

Comment:
@gammazero could you please, clarify this part? When there are multiple connections, is KeyStore instance created per each client?

Reply:
A KeyStore is provided to an authenticator instance, which is registered with the router.  All connections using the same authenticator will use the same KeyStore.  If you are trying to determine which cookie to retrieve for comparison, you need to look at authid or other information in the HELLO.Details (possibly even request data).

Each connection is given a tracking cookie with a unique value.  If the same client makes a connection in the future it can identify itself with a previously issued tracking cookie.  This means that the authenticator needs to store the nextCookie when the client is authenticated.  Then, in the future, the authenticator can see if the saved cookie matches the one presented by the client.  If the client is authenticated, then store the new nextCookie.

Reply:
> The test just looks at a single saved cookie to make sure stuff works.

Ah, sorry, completely forgot this was a test file :D 

So this line is meant for comparing stored cookie with the one from request. Thanks!